src/ratelimit.ts

```diff
@@ -16,54 +16,65 @@ export const RateLimiter = (
     const { ip } = ctx.request;
     const timestamp = Date.now();
 
+    const GET = await opt.store.get(ip)!
+    const SET = async (ip: string, options: Ratelimit) => {
+      await opt.store.set(ip, {
+        remaining: options.remaining,
+        lastRequestTimestamp: options.lastRequestTimestamp,
+      });
+    }
+    const DELETE = await opt.store.delete(ip)
+    const HAS = await opt.store.has(ip)
+
+
     if (await opt.skip(ctx)) return next();
     if (opt.headers) {
       ctx.response.headers.set("X-RateLimit-Limit", opt.max.toString());
     }
 
     if (
-      opt.store.has(ip) &&
-      timestamp - opt.store.get(ip)!.lastRequestTimestamp > opt.windowMs
+      HAS &&
+      timestamp - GET.lastRequestTimestamp > opt.windowMs
     ) {
-      opt.store.delete(ip);
+      DELETE;
     }
-    if (!opt.store.has(ip)) {
-      opt.store.set(ip, {
+    if (!HAS) {
+      await SET(ip, {
```

Comment:
This could directly be `await opt.store.set`. The provision for not using `await` inside is only enforced for the conditional part of the statement, not the execution part for when the condition is met.

i.e.

```js
if(THIS SHOULD NOT BE AWAIT) {
    THIS CAN BE AWAIT
}
```